src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/LocalizationModelDeserializer.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Localization;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects
+{
+    internal static class LocalizationModelDeserializer
+    {
+        private const string PostActionIndexPrefix = "postActions[";
+        private const string PostActionIndexSuffix = "]";
+
+        private const string ManualInstructionIndexPrefix = "manualInstructions[";
+        private const string ManualInstructionIndexSuffix = "]";
+
+        public static ILocalizationModel Deserialize(JObject data)
+        {
+            var locModel = new LocalizationModel();
+            var parameterLocalizations = new Dictionary<string, ParameterSymbolLocalizationModel>();
+
+            // Property names are in format: symbols.framework.choices.0.description
+            // Split them using '.' and store together with the localized string (property value).
+            IEnumerable<(IEnumerable<string> nameParts, string localizedString)> stringsWithNames = data.Properties()
+                .Where(p => p.Value.Type == JTokenType.String)
```

Comment:
I think I would prefer we throw some exception if this happens rather then silently swallow invalid json and let author struggle figuring it out...

Reply:
How about logging a warning? At least the user can continue to use the template until the author can fix it.

Reply:
I plan to add some more logs for other problems as well.

Reply:
I think that if we just throw exception.... template author should be able to notice this mistake before template is released.

Reply:
We throw an exception now, with proper error string. It will be caught and printed.